src/canary.py

```diff
@@ -83,33 +81,46 @@ async def watch_events(*args, **kwargs):
                 else:
                     statuses = monitor["spec"]["status"]
                 tasks[name] = asyncio.create_task(
-                    monitor_url(name, url, interval, statuses)
-                )
+                    monitor_url(name, url, interval, statuses))
+            await asyncio.gather(*tasks.values())
 
         # Consume events
         # TODO Subscript to kubes event queue for changes to CanaryHTTPMonitor objects that are visible
-        # while True:
-        #     # Get event (simulated)
-        #     await asyncio.sleep(30)
-        #     monitor = random.choice(monitors)
-        #     name = monitor["name"]
-        #     url = monitor["url"]
-        #     statuses = monitor["expect"]["status"]
-        #     interval = monitor["interval"] + random.randint(0, 10)
-        #     event = "UPDATED"
-        #
-        #     # Cancel the task if it already exists
-        #     if name in tasks:
-        #         logging.info(f"cancelling monitor [{name=}]")
-        #         tasks[name].cancel()
-        #         await tasks[name]
-        #
-        #     # Create a new task at the desired interval
-        #     if event in ["ADDED", "UPDATED"]:
-        #         logging.info(f"spawning monitor [{name=}]")
-        #         tasks[name] = asyncio.create_task(
-        #             monitor_url(name, url, interval, statuses)
-        #         )
+        watch = kubernetes_asyncio.watch.Watch()
+
+        # TODO Get namespace from pod
+        namespace = "canary"
+
+        while True:
+            logging.info("listening for streamed events")
+            async with watch.stream(crds.list_namespaced_custom_object, "canary.ukserp.ac.uk", "v1", namespace, "canaryhttpmonitors") as stream:
```

Comment:
This is listing based on namespaced objects. The above query was listing cluster objects.

The CRD is currently namespaced. Can you confirm by looking in lens what the CRD you have deployed is?

Reply:
Namespaced to ```canary``